oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,140 @@
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.jose.jws.JwsAlgorithms;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtVerifier;
+
+import javax.crypto.SecretKey;
+import java.security.PublicKey;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+public class OAuth2ResourceServerConfigurer<B extends HttpSecurityBuilder<B>> extends
+		AbstractHttpConfigurer<JwtVerifierConfigurer<B>, B> {
+
+	public BearerConfigurer bearer() {}
+
+	public class BearerConfigurer {
```

Comment:
Let's simplify this to `http.oauth2().resourceServer().bearerTokenResolver(BearerTokenResolver bearerTokenResolver)`

If the user wants to enable form body resolving than they can do this:

```
DefaultBearerTokenResolver resolver = new DefaultBearerTokenResolver();
resolver.setUseFormEncodedBodyParameter(true)

http.oauth2().resourceServer().bearerTokenResolver(resolver)
```

Reply:
Okay, sure.